addons/website_version/models/website.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import json
+from openerp import fields, models, api
+from openerp.http import request
+import random
+
+
+class NewWebsite(models.Model):
+
+    _inherit = "website"
+
+    tuto_google_analytics = fields.Boolean("How to get GA key and View ID")
+    google_analytics_view_id = fields.Char('View ID')
+    google_management_authorization = fields.Char('Google authorization')
+
+    @api.model
+    def get_current_version(self, context=None):
+        snap = request.env['website_version.version']
+        version_id = request.context.get('version_id')
+
+        if not version_id:
+            request.context['version_id'] = 0
+            return (0, '')
+        return (version_id, snap.browse(version_id).name)
```

Comment:
Why not returning a browse record ? By the way, why not a function field on website model instead of a method called directly from the templates ? Because it seems to basically be equal to a function field, except that it is not.